src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs

```diff
@@ -182,10 +183,17 @@ public async Task<TransactionPageWithoutTotal> GetTransactionStream(TransactionS
             ? request.Cursor?.StateVersionBoundary ?? request.FromStateVersion
             : request.FromStateVersion;
 
-        var searchQuery = _dbContext
-            .LedgerTransactionMarkers
-            .Where(lt => lt.StateVersion <= upperStateVersion && lt.StateVersion >= (lowerStateVersion ?? lt.StateVersion))
-            .Select(lt => lt.StateVersion);
+        var baseQuery = _dbContext
+            .LedgerTransactions
+            .Where(lt => lt.StateVersion <= upperStateVersion && lt.StateVersion >= (lowerStateVersion ?? lt.StateVersion));
+
+        var searchQuery = request.SearchCriteria.Status switch
+        {
+            LedgerTransactionStatusFilter.All => baseQuery.Select(lt => lt.StateVersion),
+            LedgerTransactionStatusFilter.Successful => baseQuery.Where(x => x.ReceiptStatus == LedgerTransactionStatus.Succeeded).Select(lt => lt.StateVersion),
```

Comment:
Out of interest, do we have indices specific to success state success/fail? 

My slight worry is that if not, we might hit the heap which might be slow because each ledger transaction row is massive.

Reply:
you're right we need that index. fixed.